src/osd/PrimaryLogPG.cc

```diff
@@ -6400,6 +6400,7 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	        oi.size - op.extent.truncate_size);
 	      ctx->modified_ranges.union_of(trim);
 	      ctx->clean_regions.mark_data_region_dirty(op.extent.truncate_size, oi.size - op.extent.truncate_size);
+	      oi.clear_data_digest();
```

Comment:
i can't understand why in here, i think this code should after line 6409.
like CEPH_OSD_OP_TRUNCATE:
when truncate smaller, we should clear_data_digest;
when truncate bigger, we should clear_data_digest too.
and clear_data_digest should after update oi.size.
could someone explain that?